### ISSUE TITLE
 avoid insert empty line before <!DOCTYPE >

### ISSUE THREAD

**Issue by csengh** [beautify-web/js-beautify#1635]
Is any technique to avoid insert empty line before `<!DOCTYPE html>` ?
Why is this not included in tag of "extra_liners" rule like [head,body,/html] ?

**Comment by bitwiseman** [comment]
@csengh Please provide input, expected output, and actual output.

**Comment by csengh** [comment]
### Input
```
1   <!DOCTYPE HTML>
2   <html>
3   <head>
4   <style>
5   </style>
6   </head>
7   <body>
8   </body>
9   </html>
```
### Expected Output

```
01   <!DOCTYPE HTML>
02   <html>
03   
04   <head>
05      <style>
06      </style>
07   </head>
08   
09   <body>
10   
11   </body>
12   
13   </html>
```
### Actual Output

```
01      <!--avoid insert this line-->
02   <!DOCTYPE HTML>
03   <html>
04   
05   <head>
06      <style>
07      </style>
08   </head>
09   
10   <body>
11   
12   </body>
13   
14   </html>
```

**Comment by bitwiseman** [comment]
@csengh 
When I use your input on https://beautifier.io/ I do not see the extra line before `<!DOCTYPE>`.  What tool are you using that you see this happening?

**Comment by csengh** [comment]
[https://beautifier.io/](url)  does not seem to work for html code:
### Input
```
<!DOCTYPE HTML>
<html>
<head>
</head>
<body>
</body>
</html>
```
### Output
![beautifier](https://user-images.githubusercontent.com/11718720/53870419-dd64e500-4002-11e9-8c27-72580f9d7e89.jpg)

**Comment by bitwiseman** [comment]
@csengh 
There is drop down that is blank. Please change that to "Beautify HTML".

**Comment by csengh** [comment]
Oh Yeah, I found it!
now online Output is OK (as expected), but unfortunately in my computer the Output is:

```
           <--this line doesn't appear in Online version-->
<!DOCTYPE HTML>
<html>

<head>
    <style>
    </style>
</head>

<body>
</body>

</html>
```

I downloaded the latest js-beautify-master.zip (CRC32: 078E2A81), unpacked and I ran::
`html-beautify -f input.html -o out.html`
the **out.html** is the one specified above, withthe  inserted empty line before <!DOCTYPE >

**Comment by bitwiseman** [comment]
I'm not able to reproduce that.  I'm not running on Windows, I'm running on macOS.  Doubt that is the cause though.  What version of node do you have installed?

**Comment by csengh** [comment]
v10.15.0
Any advice that I could try ?

**Comment by bitwiseman** [comment]
Perhaps we could add a test and see if it fails on Windows in the CI system.

**Comment by HanabishiRecca** [comment]
This happens because of UTF-8 BOM header. Encode your file to UTF-8 without BOM.

**Comment by bitwiseman** [comment]
@HanabishiRecca  How would we do this?

**Comment by csengh** [comment]
>  This happens because of UTF-8 BOM header.

I tried UTF-8 without BOM, checked with Notepad ++ but there is no change, the extra line is added :(

**Comment by HanabishiRecca** [comment]
@csengh, hmm, I can reproduce this only with BOM.
[jbtest.zip](https://github.com/beautify-web/js-beautify/files/2991194/jbtest.zip)

@bitwiseman, I think this issue caused by some internal Node.js behavior.

**Comment by bitwiseman** [comment]
@csengh 
Could you open the output file in a hex editor and send the first 20 or so bytes?  That might help us figure out what is going.

**Comment by csengh** [comment]
### input
`3c 21 44 4f 43 54 59 50 45 20 48 54 4d 4c 3e 0d 0a 3c 68 74 6d 6c 3e 0d 0a 3c 68 65 61 64 3e 0d 0a 3c 73 74 79 6c 65 3e 0d 0a 3c 2f 73 74 79 6c 65 3e 0d 0a 3c 2f 68 65 61 64 3e 0d 0a 3c 62 6f 64 79 3e 0d 0a 3c 2f 62 6f 64 79 3e 0d 0a 3c 2f 68 74 6d 6c 3e  `

### output
`0d 0a 3c 21 44 4f 43 54 59 50 45 20 48 54 4d 4c 3e 0d 0a 3c 68 74 6d 6c 3e 0d 0a 3c 68 65 61 64 3e 0d 0a 20 20 20 3c 73 74 79 6c 65 3e 0d 0a 20 20 20 3c 2f 73 74 79 6c 65 3e 0d 0a 3c 2f 68 65 61 64 3e 0d 0a 3c 62 6f 64 79 3e 0d 0a 3c 2f 62 6f 64 79 3e 0d 0a 3c 2f 68 74 6d 6c 3e  `

**Comment by bitwiseman** [comment]
@csengh Hm, so it literally is just a `\r\n` at the top of the file.

**Comment by csengh** [comment]
yes, in the output file, but the input file does not have it . . .
